### DISCUSSION TITLE
Is there a way to get useForm errors on custom resolver?

### DISCUSSION THREAD

**Discussion by criticalx7** [react-hook-form/react-hook-form#8847] [Q&A] [answered]
I noticed that when I set a custom error (name not registered) via `setError`, a form can still be submitted without the error.

I guess it's because the resolver swallows the error set by setError. 

## Question 
Is there a way to get the original form error in resolver ? So I can merge it with the resolvers before it hits `handleSubmit` ?

## Note:
1. Is it because I didn't register the name first so the `handleSubmit` assumed the error was okay?
2. I uses a Zod resolver.

**Comment by Moshyfawn** [comment, marked as answer]
Check this issue out: #8836

**Reply by criticalx7** [reply]
Thanks for the info man. From what I understand,  it means I need to validate from schema only, right?  The error from setError won't count and will get swallowed if used with the resolver.

**Reply by Moshyfawn** [reply]
The custom error set via `setError` won't count, if you don't have it "attached" to a registered field.

Basically, for `setError` induced error to count towards validation, you need to set it to an existing field / created a hidden input to "attach" all your global errors to. A kind of current design limitation with RHF
